engines/wage/wage.cpp

```diff
@@ -0,0 +1,505 @@
+
+	// Your main event loop should be (invoked from) here.
+	_resManager = new Common::MacResManager();
+	_resManager->open(getGameFile());
+
+	_world = new World(this);
+
+	if (!_world->loadWorld(_resManager))
+		return Common::kNoGameDataFoundError;
+
+	_gui = new Gui(this);
+
+	_temporarilyHidden = true;
+	performInitialSetup();
+	Common::String input("look");
+	processTurn(&input, NULL);
+	_temporarilyHidden = false;
+
+	_shouldQuit = false;
+
+	while (!_shouldQuit) {
+		processEvents();
+
+		_gui->draw();
+		g_system->updateScreen();
+		g_system->delayMillis(50);
+	}
+
+	//_world->_orderedScenes[1]->_design->paint(&screen, _world->_patterns, false);
+	//_world->_objs["frank.1"]->_design->setBounds(&r);
+	//_world->_objs["frank.1"]->_design->paint(&screen, _world->_patterns, false);
+	//_world->_scenes["temple of the holy mackeral"]->_design->setBounds(&r);
+	//_world->_scenes["temple of the holy mackeral"]->_design->paint(&screen, _world->_patterns, false);
+	//_world->_scenes["tower level 3"]->_design->setBounds(&r);
+	//_world->_scenes["tower level 3"]->_design->paint(&screen, _world->_patterns, false);
+
+	return Common::kNoError;
+}
+
+void WageEngine::processEvents() {
+	Common::Event event;
+
+	while (_eventMan->pollEvent(event)) {
+		switch (event.type) {
+		case Common::EVENT_QUIT:
+			if (saveDialog())
+				_shouldQuit = true;
+			break;
+		case Common::EVENT_MOUSEMOVE:
+			_gui->mouseMove(event.mouse.x, event.mouse.y);
+			break;
+		case Common::EVENT_LBUTTONDOWN:
+			_gui->mouseDown(event.mouse.x, event.mouse.y);
+			break;
+		case Common::EVENT_LBUTTONUP:
+			{
+				Designed *obj = _gui->mouseUp(event.mouse.x, event.mouse.y);
+				if (obj != NULL)
+					processTurn(NULL, obj);
+			}
+			break;
+		case Common::EVENT_KEYDOWN:
+			switch (event.kbd.keycode) {
+			case Common::KEYCODE_BACKSPACE:
+				if (_inputText.size()) {
+					_inputText.deleteLastChar();
+					_gui->drawInput();
+				}
+				break;
+
+			case Common::KEYCODE_RETURN:
+				if (_inputText.empty())
+					break;
+
+				processTurn(&_inputText, NULL);
+				_inputText = "";
+				_gui->appendText("");
+				_gui->disableUndo();
+				break;
+
+			default:
+				if (event.kbd.flags & (Common::KBD_ALT | Common::KBD_CTRL | Common::KBD_META)) {
+					if (event.kbd.ascii >= 0x20 && event.kbd.ascii <= 0x7f) {
+						_gui->processMenuShortCut(event.kbd.flags, event.kbd.ascii);
+					}
+					break;
+				}
+
+				if (event.kbd.ascii >= 0x20 && event.kbd.ascii <= 0x7f) {
+					_inputText += (char)event.kbd.ascii;
+					_gui->drawInput();
+				}
+
+				break;
+			}
+			break;
+
+		default:
+			break;
+		}
+	}
+}
+
+void WageEngine::playSound(Common::String soundName) {
+	warning("STUB: WageEngine::playSound(%s)", soundName.c_str());
+}
+
+void WageEngine::setMenu(Common::String menu) {
+	_world->_commandsMenu = menu;
+
+	_gui->regenCommandsMenu();
+}
+
+void WageEngine::appendText(const char *str) {
+	_gui->appendText(str);
+
+	if (_inputText.size())
+		_inputText = "";
```

Comment:
This can be simplified to `_inputText.clear()` and the `size` (again really a `!empty` check) seems superflous.